src/main/java/edu/ie3/datamodel/models/input/container/SystemParticipants.java

```diff
@@ -404,6 +404,11 @@ public SystemParticipantsCopyBuilder storages(Set<StorageInput> storages) {
      */
     public SystemParticipantsCopyBuilder wecPlants(Set<WecInput> wecPlants) {
       this.wecPlants = wecPlants;
+      return (SystemParticipantsCopyBuilder) thisInstance();
+    }
+
+    @Override
+    protected InputContainerCopyBuilder<SystemParticipantInput> thisInstance() {
```

Comment:
You should set the returned type to `SystemParticiapntCopyBuilder` and them remove the type casts.